x/emissions/keeper/inference_synthesis/network_inference_builder.go

```
@@ -218,189 +355,503 @@ func (b *NetworkInferenceBuilder) SetOneOutInfererValues() *NetworkInferenceBuil
+	oneOutNetworkInferenceWithoutInferer alloraMath.Dec,
+	err error,
+) {
+	args.Logger.Debug(fmt.Sprintf("Calculating one-out inference for topic %v withheld forecaster %s", args.TopicId, args.WithheldForecaster))
 
 	// Remove the withheldForecaster from the palette's forecasters
-	remainingForecasters := make([]Worker, 0)
-	for _, forecaster := range palette.Forecasters {
-		if forecaster != withheldForecaster {
+	remainingForecasters := make([]Forecaster, 0)
+	remainingForecasterToForecast := make(map[Forecaster]*emissions.Forecast)
+	remainingForecasterRegrets := make(map[Forecaster]*Regret)
+	for _, forecaster := range args.Forecasters {
+		if forecaster != args.WithheldForecaster {
 			remainingForecasters = append(remainingForecasters, forecaster)
-		}
-	}
 
-	err := palette.UpdateForecastersInfo(remainingForecasters)
-	if err != nil {
-		return alloraMath.Dec{}, errorsmod.Wrapf(err, "Error updating forecasters")
-	}
+			regret, _, err := args.K.GetOneOutForecasterForecasterNetworkRegret(args.Ctx, args.TopicId, args.WithheldForecaster, forecaster)
+			if err != nil {
+				return alloraMath.Dec{}, errorsmod.Wrapf(err, "calcOneOutForecasterInference() error getting one-out forecaster regret")
+			}
+			remainingForecasterRegrets[forecaster] = &regret.Value
 
-	// Get regrets for the remaining inferers
-	palette.InfererRegrets = make(map[string]*alloraMath.Dec)
-	for _, inferer := range palette.Inferers {
-		regret, _, err := palette.K.GetOneOutForecasterInfererNetworkRegret(palette.Ctx, palette.TopicId, withheldForecaster, inferer)
-		if err != nil {
-			return alloraMath.Dec{}, errorsmod.Wrapf(err, "Error getting one-out inferer regret")
+			forecast, ok := args.ForecasterToForecast[forecaster]
+			if !ok {
+				args.Logger.Debug(fmt.Sprintf("Cannot find forecaster in ForecasterRegrets in UpdateForecastersInfo %v", forecaster))
+				continue
+			}
+			remainingForecasterToForecast[forecaster] = forecast
```

I think using `continue` here will break stuff in further calculations; please double-check if it's not better to just throw an error.

---

that's taken from this line here:

https://github.com/allora-network/allora-chain/blob/fd6486788fb2f37d78ce23281d7296c89586753f/x/emissions/keeper/inference_synthesis/synth_palette_weight.go#L264-L281

so it would be a change from what `dev` does. I am not against throwing an error here instead of continuing, but want to note that

---

made the change, I agree it should be an error not skipped